hyperscan fails to install due to missing boost files

### Xmake Version

xmake v3.0.3+20250922

### Operating System Version and Architecture

Linux Cuddles-Bobba 6.16.7-arch1-1 #1 SMP PREEMPT_DYNAMIC Thu, 11 Sep 2025 17:42:36  +0000 x86_64 GNU/Linux

### Describe Bug

`xrepo install hyperscan` fails due to missing boost files:

```
/home/dkaszews/.xmake/cache/packages/2509/h/hyperscan/v5.4.2/source/src/util/small_vector.h:46:12: fatal error: boost/container/small_vector.hpp: No such file or directory
   46 | #  include <boost/container/small_vector.hpp>
      |            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Exact missing file reported depends on the compilation order, but I have grepped all includes in the source directory, and compared it to the list of files in include path and came up with the following:

```
boost/container/small_vector.hpp
boost/graph/adjacency_iterator.hpp
boost/graph/adjacency_list.hpp
boost/graph/boykov_kolmogorov_max_flow.hpp
boost/graph/breadth_first_search.hpp
boost/graph/connected_components.hpp
boost/graph/copy.hpp
boost/graph/dag_shortest_paths.hpp
boost/graph/depth_first_search.hpp
boost/graph/filtered_graph.hpp
boost/graph/graph_traits.hpp
boost/graph/properties.hpp
boost/graph/property_maps/constant_property_map.hpp
boost/graph/reverse_graph.hpp
boost/graph/strong_components.hpp
boost/graph/topological_sort.hpp
boost/graph/transpose_graph.hpp
boost/pending/property.hpp
boost/random.hpp
boost/random/mersenne_twister.hpp
boost/random/uniform_int_distribution.hpp
```

I believe this is caused by missing `{ config = { container = true, graph = true, random = true } }` in `add_deps('boost')`, but don't know how to pull in the `boost/pending/property.hpp` as `pending` is not present in https://github.com/xmake-io/xmake-repo/blob/dev/packages/b/boost/libs.lua .

Running `xmake install -f 'container=true,random=true' boost` does install most of the missing files, but `xmake install -f 'graph=true' boost` fails with another error:

```
CMake Error at libs/algorithm/CMakeLists.txt:15 (target_link_libraries):
  The link interface of target "boost_algorithm" contains:

    Boost::exception

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.



CMake Error at libs/lexical_cast/CMakeLists.txt:15 (target_link_libraries):
  The link interface of target "boost_lexical_cast" contains:

    Boost::container

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.



CMake Error at libs/spirit/CMakeLists.txt:15 (target_link_libraries):
  The link interface of target "boost_spirit" contains:

    Boost::thread

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.



CMake Error at libs/xpressive/CMakeLists.txt:15 (target_link_libraries):
  The link interface of target "boost_xpressive" contains:

    Boost::exception

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.
```

### Expected Behavior

Install succeeds

### Project Configuration

N/A, run manually from command line

### Additional Information and Error Logs

https://gist.github.com/dkaszews/b1c6634097af483db9055b5a2c655962

--- a/packages/h/hyperscan/xmake.lua
+++ b/packages/h/hyperscan/xmake.lua
@@ -7,7 +7,13 @@ package("hyperscan")
              "https://github.com/intel/hyperscan.git")
     add_versions("v5.4.2", "32b0f24b3113bbc46b6bfaa05cf7cf45840b6b59333d078cc1f624e4c40b2b99")
 
-    add_deps("cmake", "boost", "ragel", "python")
+    add_deps("cmake", "ragel", "python")
+    add_deps("boost", {configs = {
+        exception = true,
+        container = true,
+        thread = true,
+        graph = true
+    }})
 
     on_install("linux", "windows|!arm*", function (package)
         local configs = {}

--- a/packages/h/hyperscan/xmake.lua
+++ b/packages/h/hyperscan/xmake.lua
@@ -8,12 +8,7 @@ package("hyperscan")
     add_versions("v5.4.2", "32b0f24b3113bbc46b6bfaa05cf7cf45840b6b59333d078cc1f624e4c40b2b99")
 
     add_deps("cmake", "ragel", "python")
-    add_deps("boost", {configs = {
-        exception = true,
-        container = true,
-        thread = true,
-        graph = true
-    }})
+    add_deps("boost", {configs = {thread = true, graph = true}})
 
     on_install("linux", "windows|!arm*", function (package)
         local configs = {}

--- a/packages/h/hyperscan/xmake.lua
+++ b/packages/h/hyperscan/xmake.lua
@@ -5,14 +5,25 @@ package("hyperscan")
 
     add_urls("https://github.com/intel/hyperscan/archive/refs/tags/$(version).tar.gz",
              "https://github.com/intel/hyperscan.git")
+
     add_versions("v5.4.2", "32b0f24b3113bbc46b6bfaa05cf7cf45840b6b59333d078cc1f624e4c40b2b99")
 
-    add_deps("cmake", "ragel", "python")
+    add_deps("cmake", "ragel", "python 3.x", {kind = "binary"})
     add_deps("boost", {configs = {thread = true, graph = true}})
 
-    on_install("linux", "windows|!arm*", function (package)
-        local configs = {}
-        table.insert(configs, "-DCMAKE_BUILD_TYPE=" .. (package:debug() and "Debug" or "Release"))
+    on_check(function (package)
+        if not package:is_arch("x64", "x86", "x86_64") then
+            raise("package(hyperscan) only support x86 arch")
+        end
+    end)
+
+    on_install(function (package)
+        if not package:config("pic") then
+            io.replace("CMakeLists.txt", "POSITION_INDEPENDENT_CODE TRUE", "POSITION_INDEPENDENT_CODE FALSE", {plain = true})
+        end
+
+        local configs = {"-DBUILD_EXAMPLES=OFF"}
+        table.insert(configs, "-DCMAKE_BUILD_TYPE=" .. (package:is_debug() and "Debug" or "Release"))
         table.insert(configs, "-DBUILD_SHARED_LIBS=" .. (package:config("shared") and "ON" or "OFF"))
         import("package.tools.cmake").install(package, configs)
     end)

--- a/packages/h/hyperscan/xmake.lua
+++ b/packages/h/hyperscan/xmake.lua
@@ -12,12 +12,14 @@ package("hyperscan")
     add_deps("boost", {configs = {thread = true, graph = true}})
 
     on_check(function (package)
-        if not package:is_arch("x64", "x86", "x86_64") then
+        if not package:is_arch("x64", "x86", "x86_64", "i386") then
             raise("package(hyperscan) only support x86 arch")
         end
     end)
 
-    on_install(function (package)
+    -- mingw require this patch: https://github.com/intel/hyperscan/pull/36
+    on_install("!mingw", function (package)
+        io.replace("CMakeLists.txt", "add_subdirectory(tools)", "", {plain = true})
         if not package:config("pic") then
             io.replace("CMakeLists.txt", "POSITION_INDEPENDENT_CODE TRUE", "POSITION_INDEPENDENT_CODE FALSE", {plain = true})
         end

